### PR TITLE
search: don't depend on args for any job creation except repo job

### PR DESCRIPTION
**Pull request by rvantonder**
Stacked on https://github.com/sourcegraph/sourcegraph/pull/32808.

## Test plan
Semantics-preserving.

### CODE REVIEW

File: internal/search/job/job.go [hunk 1]
```diff
@@ -55,17 +55,19 @@ func ToSearchJob(jargs *Args, q query.Q) (Job, error) {
...
 	resultTypes := search.ComputeResultTypes(types, b.PatternString(), jargs.SearchInputs.PatternType)
 
-	args := toTextParameters(jargs, b, resultTypes)
+	patternInfo := search.ToTextPatternInfo(b, resultTypes, jargs.SearchInputs.Protocol)
+	if patternInfo.Pattern == "" {
+		// Fallback to basic search for searching repos and files if
+		// the structural search pattern is empty.
+		jargs.SearchInputs.PatternType = query.SearchTypeLiteral
+		patternInfo.IsStructuralPat = false
+	}
```
Comment (rvantonder) [inline internal/search/job/job.go:64]: Inlined

File: internal/search/job/job.go [hunk 2]
```diff
@@ -55,17 +55,19 @@ func ToSearchJob(jargs *Args, q query.Q) (Job, error) {
...
 
 	// searcher to use full deadline if timeout: set or we are streaming.
 	useFullDeadline := q.Timeout() != nil || q.Count() != nil || jargs.SearchInputs.Protocol == search.Streaming
 
 	features := toFeatures(jargs.SearchInputs.Features)
 	repoOptions := toRepoOptions(q, jargs.SearchInputs.UserSettings)
-	// explicitly populate RepoOptions field in args, because the repo search job
-	// still relies on all of args. In time it should depend only on the bits it truly needs.
-	args.RepoOptions = repoOptions
```
Comment (rvantonder) [inline internal/search/job/job.go:67]: We already take care of this below, in the RepoJob, don't need to do it here and depend on args.

File: internal/search/job/job.go [hunk 3]
```diff
@@ -340,18 +342,20 @@ func ToSearchJob(jargs *Args, q query.Q) (Job, error) {
...
 					// repoHasFile operation. It is slated
 					// for removal.
+					args := search.TextParameters{}
 					args.Zoekt = jargs.Zoekt
 					args.SearcherURLs = jargs.SearcherURLs
 					args.RepoOptions = repoOptions
 					args.Features = features
 					args.UseFullDeadline = useFullDeadline
 					args.Query = q
+					args.PatternInfo = patternInfo
```
Comment (rvantonder) [inline internal/search/job/job.go:358]: This looks silly, just mechanical.

**Review by camdencheek — APPROVED** [review]

File: internal/search/job/job.go (hunk 3 above)
Comment (camdencheek) [reply internal/search/job/job.go:358]: It made the diff nice and readable, so I'm not complaining

**Pull request merged**